content/terraform/v1.12.x/docs/language/block/provider.mdx

```diff
@@ -0,0 +1,419 @@
+- Required: Varies by provider
+
+### `alias`
+
+Optionally use the `alias` argument to define multiple configurations for the same provider. Defining multiple provider aliases lets you specify which provider configuration to use for individual resources, data sources, or modules.
+
+To create multiple configurations for a given provider, include multiple `provider` blocks with the same provider name, then add the `alias` argument to each additional provider configuration to give it a unique identifier.
+
+<CodeBlockConfig highlight="6" hideClipboard>
+
+```hcl
+provider "exampleName" {
+  region = "us-east-1"
+}
+
+provider "exampleName" {
+  alias  = "west"
+  region = "us-west-1"
+}
+```
+
+</CodeBlockConfig>
+
+To refer to a provider alias, use `<PROVIDER_NAME>.<ALIAS>` in the `provider` argument of the following blocks:
+
+- [`resource` blocks](/terraform/language/block/resource)
+- [`data` blocks](/terraform/language/block/data)
+- [`module` blocks](/terraform/language/modules/syntax)
+
+If there are multiple aliases for a provider, the `provider` block without an `alias` argument is the default configuration for that provider. Resources, data sources, and modules that don't specify the [`provider` meta-argument](/terraform/language/block/resource#provider) use the default provider configuration that matches the resource type name.
+
+#### Summary
+
+- Data type: String
+- Default: None
+- Refer to the following examples to learn more about aliasing providers:
+  - [Multiple provider configurations](#multiple-provider-configurations)
+  - [Using an alternate provider configuration](#using-an-alternate-provider-configuration)
+  - [Provider without a default configuration](#provider-without-a-default-configuration)
+  - [Using an alternate provider configuration in a child module](#using-an-alternate-provider-configuration-in-a-child-module)
+
+### `version`
+
+The `version` argument in provider configurations is deprecated, and Terraform will remove it in a future version. Instead, declare provider version constraints in [the `terraform` block's `required_providers` argument](/terraform/language/providers/requirements).
+
+#### Summary
+
+- Data type: String
+- Default: None
+- Deprecated
+
+## Examples
+
+The following examples demonstrate common use cases for `provider` blocks.
+
+### Basic provider configuration
+
+The following example configures the `google` provider with a specific project and region. In the `terraform` block's `required_providers` block you define the provider version you want to use, where to source the provider from, and the local name of the provider:
+
+<CodeBlockConfig filename="terraform.tf">
+
+```hcl
+terraform {
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = "~> 4.0"
+    }
+  }
+}
+```
+
+</CodeBlockConfig>
+
+You can further configure a provider with the `provider` block in the root module of your configuration.
```

Comment:
```suggestion
You can configure the provider with the `provider` block in the root module of your configuration.
```

Maybe I'm splitting hairs, but my thinking is that the `required_providers` block configures *terraform* to use the specified provider, and the *provider's* configuration is in the provider block?

Reply:
Also...should we mention somewhere that the `terraform { }` block being in `terraform.tf` and the `provider "google" { }` block in `main.tf` is our recommended convention?

Reply:
Oooh good point! I think the location of the blocks will be what we cover in the usage page though